lib/src/utils/formatter_inputs.dart

```diff
@@ -0,0 +1,75 @@
+import 'dart:io';
+
+import 'package:glob/glob.dart';
+import 'package:path/path.dart' as p;
+
+/// Builds and returns the object that contains:
+/// - The file paths
+/// - The paths that were excluded by an exclude glob
+/// - The paths that were skipped because they are links
+/// - The hidden directories(start with a '.') that were skipped
+///
+/// The file paths will include all .dart files in [path] (and its subdirectories),
+/// except any paths that match the expanded [exclude] globs.
+///
+/// By default these globs are assumed to be relative to the current working
+/// directory, but that can be overridden via [root] for testing purposes.
+FormatterInputs getFormatterInputs({List<Glob> exclude, String root}) {
+  FormatterInputs inputs = FormatterInputs(Set(), Set(), Set(), Set());
+
+  exclude ??= <Glob>[];
+
+  if (exclude.isEmpty) inputs.includedFiles.add(root ?? '.');
```

Comment:
If there are no excludes, we should bail out early and return just the directory. Also, I told you wrong earlier when you asked about this, in the case where `root` is not null, we should still return `.` – that is because the executable will be run from that same `root` path, so we want the paths to be relative to that dir.
```dart
if (exclude.isEmpty) {
  return FormatterInputs({'.'});
}
```